engine/include/cubos/engine/ui/canvas/element.hpp

```diff
@@ -20,24 +23,32 @@ namespace cubos::engine
     struct CUBOS_ENGINE_API UIElement
     {
         CUBOS_REFLECT;
-
         glm::vec2 offset = {0.0F, 0.0F}; ///< Offset of the pivot.
         glm::vec2 size = {0.0F, 0.0F};   ///< Size of the element.
         glm::vec2 pivot = {0.5F, 0.5F};  ///< Relative position of the pivot to the element itself.
         glm::vec2 anchor = {0.5F, 0.5F}; ///< Relative position of the pivot to the parent element.
         int layer = 0;                   ///< Rendering order layer.
 
         glm::vec2 position = {0.0F, 0.0F}; ///< Global position of the pivot.
-        glm::mat4 vp = {};                 ///< View matrix provided by the canvas.
 
         int hierarchyDepth = 0; ///< Depth in hierarchy. Used for depth sorting.
 
+        UIDrawList drawList; ///< Draw list holding draw commands from this element.
+
         /// @brief Gets global position of the bottom left corner of the element.
         /// @return Position value in global coordinates.
         glm::vec2 bottomLeftCorner() const;
 
         /// @brief Gets global position of the top right corner of the element.
         /// @return Position value in global coordinates.
         glm::vec2 topRightCorner() const;
+
+        /// @brief Adds entry to draw list.
+        /// @param entry Entry to add.
+        /// @param data Pointer to per element data.
+        void pushEntry(UIDrawListEntry entry, const void* data);
+
+        /// @brief Empties draw list.
+        void clearCommands();
```

Comment:
No need to have this method as the canvas will access the drawList directly anyway I think.